Core.Collectors/Web/HttpResponseSignature.cs

```diff
@@ -13,17 +16,34 @@ public class HttpResponseSignature
         private readonly HttpStatusCode statusCode;
         private readonly Regex responseMessageRegex;
         private readonly Func<JObject, bool> matcher;
+        private readonly Func<Task> handler;
 
-        public HttpResponseSignature(HttpStatusCode statusCode, string responseMessageRegex)
+        public HttpResponseSignature(HttpStatusCode statusCode, string responseMessageRegex, Func<Task> handler = null)
         {
             this.statusCode = statusCode;
             this.responseMessageRegex = new Regex($"^{responseMessageRegex}$", RegexOptions.Compiled | RegexOptions.Multiline);
+            this.handler = handler;
         }
 
-        public HttpResponseSignature(HttpStatusCode statusCode, Func<JObject, bool> matcher)
+        public HttpResponseSignature(HttpStatusCode statusCode, Func<JObject, bool> matcher, Func<Task> handler = null)
         {
             this.statusCode = statusCode;
             this.matcher = matcher;
+            this.handler = handler;
+        }
+
+        public async Task<bool> Matches(HttpResponseMessage response, string responseMessagePath = "$.message")
+        {
+            try
+            {
+                string responseContent = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+                JObject responseContentObject = JObject.Parse(responseContent);
+                return this.Matches(response.StatusCode, responseContentObject, responseMessagePath);
+            }
+            catch (JsonReaderException)
+            {
+                return false;
+            }
         }
```

Comment:
Why do we need this? Can't we parse the response into the JObject in the caller and then use the existing method?